runtime/service/src/main/java/org/apache/polaris/service/task/TableCleanupTaskHandler.java

```diff
@@ -71,21 +71,19 @@ public TableCleanupTaskHandler(
 
   @Override
   public boolean canHandleTask(TaskEntity task) {
-    return task.getTaskType() == AsyncTaskType.ENTITY_CLEANUP_SCHEDULER && taskEntityIsTable(task);
+    return task.getTaskType() == AsyncTaskType.ENTITY_CLEANUP_SCHEDULER
+        && tryGetTableEntity(task).isPresent();
   }
 
-  private boolean taskEntityIsTable(TaskEntity task) {
-    PolarisEntity entity = PolarisEntity.of((task.readData(PolarisBaseEntity.class)));
-    return entity.getType().equals(PolarisEntityType.TABLE_LIKE);
+  private Optional<IcebergTableLikeEntity> tryGetTableEntity(TaskEntity task) {
```

Comment:
nit: the "try" prefix makes me wonder whether the method may or may not produce an "effect" (as in `Semaphore.tryAcquire()`), but in fact it always operates the same way without side effects... I'd prefer something like `asTableEntity(TaskEntity task)` or `asOptionalTableEntity()`.